code-challenges/for-each/challenges-01.test.js

```diff
@@ -0,0 +1,230 @@
+
+Write a function named greeting that takes in a string and returns the string in all uppercase letters.
+
+Then, write a function named speaker that takes in a string and a callback function. 
+The speaker function should return the string in all uppercase letters only by invoking the callback.
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+    const result = word.toUpperCase();
+    return result;
+};
+
+const speaker = (message, callback) => {
+    return callback(message);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. 
+This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third 
+argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+    arr.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+    for (let i = 0; i < times; i++) {
+        callback(arr, num);
+    }
+    return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named removeOne that takes in a number and an array. 
+If the number divided by three has a remainder of two, pop one element off of the array.
+
+Hint: you may want to look into the modulo operation.
+
+Then, write a function named removeElements that takes in an array and a callback. 
+This function should use a for loop to iterate over the array and invoke the callback once for each element 
+in the array.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const removeOne = (num, arr) => {
+    if (num % 3 === 2) {
+        arr.pop();
+    }
+};
+
+const removeElements = (arr, callback) => {
+    for (let i = 0; i < arr.length; i++) {
+        callback(arr[i], arr);
+    }
+    return arr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named removeWithForEach that produces the same output as challenge 3, but uses forEach.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithForEach = (arr, callback) => {
+    arr.forEach((value, idx) => {
```

Comment:
`idx` in this case in an optional param, if youre not using in in your code, go ahead and leave it out